src/native/shared/perfetto_support.hh

```diff
@@ -0,0 +1,195 @@
+		static constexpr std::string_view MethodName_AnnotationName   { "Method name" };
+		static constexpr std::string_view ClassName_AnnotationName    { "Class name" };
+
+		static constexpr std::string_view Null_AnnotationContent      { "<NULL>" };
+		static constexpr std::string_view MissingMethodName           { "<UNNAMED METHOD>" };
+
+	public:
+
+		force_inline constexpr static perfetto::StaticString get_event_name (std::string_view const& sv)
+		{
+			return perfetto::StaticString { sv.data () };
+		}
+
+		template<xamarin::android::PerfettoTrackId TTrack, bool UseThreadTrack>
+		force_inline static perfetto::Track get_name_annotated_track ()
+		{
+			using TParentTrack = std::conditional_t<UseThreadTrack, perfetto::ThreadTrack, perfetto::ProcessTrack>;
+			auto track = perfetto::Track (static_cast<uint64_t>(PerfettoTrackId::MonodroidRuntime), TParentTrack::Current ());
+			auto desc = track.Serialize ();
+
+			// if constexpr (TTrack == PerfettoTrackId::AssemblyLoadMonoVM) {
+			// 	desc.set_name (PerfettoConstants::AssemblyLoadAnnotation.data ());
+			// } else if constexpr (TTrack == PerfettoTrackId::ImageLoadMonoVM) {
+			// 	desc.set_name (PerfettoConstants::ImageLoadAnnotation.data ());
+			// } else if constexpr (TTrack == PerfettoTrackId::ClassLoadMonoVM) {
+			// 	desc.set_name (PerfettoConstants::ClassLoadAnnotation.data ());
+			// } else if constexpr (TTrack == PerfettoTrackId::VTableLoadMonoVM) {
+			// 	desc.set_name (PerfettoConstants::VTableLoadAnnotation.data ());
+			// } else if constexpr (TTrack == PerfettoTrackId::MethodInvokeMonoVM) {
+			// 	desc.set_name (PerfettoConstants::MethodInvokeAnnotation.data ());
+			// } else if constexpr (TTrack == PerfettoTrackId::MethodInnerMonoVM) {
+			// 	desc.set_name (PerfettoConstants::MethodRunTimeAnnotation.data ());
+			// } else if constexpr (TTrack == PerfettoTrackId::MonitorContentionMonoVM) {
+			// 	desc.set_name (PerfettoConstants::MonitorContentionAnnotation.data ());
+			// } else if constexpr (TTrack == PerfettoTrackId::MonodroidRuntime) {
+				desc.set_name (PerfettoConstants::MonodroidRuntimeTrack.data ());
+			//}
+
+			set_track_event_descriptor (track, desc);
+			return track;
+		}
+
+		template<xamarin::android::PerfettoTrackId TTrack>
+		[[gnu::flatten]]
+		force_inline static perfetto::Track get_name_annotated_process_track ()
+		{
+			return get_name_annotated_track<TTrack, false> ();
+		}
+
+		template<xamarin::android::PerfettoTrackId TTrack>
+		[[gnu::flatten]]
+		force_inline static perfetto::Track get_name_annotated_thread_track ()
+		{
+			return get_name_annotated_track<TTrack, true> ();
+		}
+
+		template<detail::SupportedMonoType TMonoType>
+		force_inline static void add_name_annotation (perfetto::EventContext &ctx, TMonoType *data)
+		{
+			std::string name{};
+			const std::string_view *annotation_name = nullptr;
+
+			if constexpr (std::same_as<MonoAssembly, TMonoType>) {
```

Comment:
One wonders why method overloads aren't valid/possible…

Reply:
They are both valid and possible, I just prefer this as a more compact form of the code.  All the possible specializations are related here, are treated in the end in the same way and this way the code is drier than with overloads and easier to follow.